### DISCUSSION TITLE
Bestway Pump model changes for 2022

### DISCUSSION THREAD

**Discussion by JPMNZ** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#431] [Ideas]
The Table of Bestway Spas says that the Tahiti model is eggshaped and should have model.h set to PRE2021.
Having acquired a Tahiti late 2022, this new pump is now square and the software works with the PCB_2, DEVICE=MIAMI2021and the 8-port YF08E chip (sub for TXS0108E).

It is also good practice to pop a few capacitors on the supply rails to reduce interference (spikes, etc) so I've put 100uF and 0.1uF on each of the 5v and 3,3v supply rails. There are sufficient adjacent solder pads on the V2 board to do this.

Delighted with the outcome!

**Comment by franpalacio** [comment]
Tahiti model with the S100101 pump.
Hello, I think I have that model but I have not been able to make it work.
What I get is that the welcome music plays, shows the IP address and date on the pump screen, and it doesn't show anything else. The screen remains half lit and the buttons do not respond, as shown in the photo.
The blue led of the esp8266 keeps blinking rapidly.
The Web interface can be accessed but the temperature shows 0 and the rest of the controls do not work.
I have already tried everything, now I have seen your post and I have noticed that the chip had and it had the 8-port YF08E, I have also tried to put the capacitors that you mention, I understand that you mean 100uF between 5V and GND (I have put it to the input of the 6 wires), and 0.1uF between 3.3v and GND (I have put it on the pins of the esp8266), and I have broken the esp8266.
Things I've tried and combinations:
-LLC red
-YF08E chip
-Esp8266 large and small.
-With and without resistors and capacitors in the 6 wires
-Cross the wires as in the V2B version of the circuit (I think I broke the esp8266)
-I have ordered the esp8266 and the YF08E again elsewhere.
-I have checked the continuity of the board and the welds with the tester.
-Change in models.h to MIAMI2021 and PRE2021.
I put photos of everything, I don't know what to do anymore. Any ideas?
![IMG_20230113_202619](https://user-images.githubusercontent.com/109759859/212410069-a03048f1-553b-48e8-9d4d-e9629053f5c1.jpg)
![IMG_20230113_202624](https://user-images.githubusercontent.com/109759859/212410105-36652d0b-4c79-4445-9450-a18f018a21cf.jpg)
![IMG_20230113_202644](https://user-images.githubusercontent.com/109759859/212410109-6c111b65-1989-4505-9407-185a11feb9fe.jpg)
![IMG_20230113_202717](https://user-images.githubusercontent.com/109759859/212410115-f5ef9b90-c025-4df6-86b1-aeed4fc2bae1.jpg)
![IMG_20230113_202737](https://user-images.githubusercontent.com/109759859/212410124-b55b3a40-b8c6-4afc-9d29-2be357473a29.jpg)

**Comment by franpalacio** [comment]
![IMG_20230113_202957](https://user-images.githubusercontent.com/109759859/212410766-104a50f9-2d15-41bb-bee8-880663de546a.jpg)

https://user-images.githubusercontent.com/109759859/212411108-5e9c2ddc-d56a-4822-866b-03dc5d3ff5c2.mp4

**Comment by JPMNZ** [comment]
Hi

Your Video looks like my experience after installing in the Tahiti with PRE2021 set in model.h. 

I assume you have the official PC board. The capacitors probably won’t make a huge difference, but they might help suppress power surges when the heater or pump turn off/on.

 

There was a problem with PRE2021 that made the system unworkable (I thought I’d damaged the chip), however, I then erased the flash memory (you can erase the chip via the platform) after my attempt with PRE2021 and reflashed again completely with MIAMI2021. I also did not use OTA 

 

This might assist.

Kind regards

JPM

**Comment by franpalacio** [comment]
I have already flashed with the two models, PRE2021 and MIAMI 2021. And also the file system and I have not achieved anything.
With both models I get the same behavior.
The board is the one you see in the photo, I ordered it from the downloaded schematic files.
I have also placed baseboards to be able to replace components as you can see in the photo.
Thank you anyway.

**Comment by franpalacio** [comment]
Yes, I find it curious that on the handle of the casing it says 2020-09-09.
I bought the spa in 2021, and in theory according to the table that pump model, because it does not match the spa model is MIAMI2021
By the way, it is the Tahiti model with the S100101 pump.

**Comment by JPMNZ** [comment]
Hi

Perhaps Bestway have supplied a more common pump. I was surprised to find several pool types that had the same dimensions and specifications, but different names!

 

The “egg” shaped pump appears to have been superseded and the square pump replaces it. The MIAMI2021 is the only #Def available and, for me, it works. I’ve tried the PRE2021 #def and that behaves much as your video showed. If you have modified any code, suggest refresh the code from Github and start all over again. 

 

Suggest also, test out the pathways between the plugs and the ESP (without it in place). Supply 5v to the board (at the Plug) and 3.3V to the R1 Resistor and test that the Level Shifter is functioning on all lines. Only a Voltmeter is needed. You should get 3.3v on the ESP side for each 5v supplied to the Pump side of the Level Shifter (and vice versa). If you have no Power Supplies, try three AA batteries on the 5v and two on the 3v – simply to test everything. (Keep the ESP out of the circuit!). 

 

Also, check every solder joint. A small defect can bring everything to a halt.

 

My Tahiti pump model is S100103 (who knows what the difference is between your model and mine!)

 

Best regards

**Comment by franpalacio** [comment]
Hello, the code is updated with the latest version of github, I have tried what you have told me, I have removed the esp board, and connected 3.3v to the 3.3 pin where the esp would go, and 5v to the 6-wire socket, then I have connected the GND.
He does not understand the result, without connecting any other cable, in the level converter board, in all the low pins it gives me 3.3v and in the high ones 5v. I have tried another level converter board and the same thing happens.
I do not know what I'm doing wrong.
What I think should happen is that in the level converter, on each low pin it would measure 3.3v when on the high side it would connect 5v and vice versa

**Comment by JPMNZ** [comment]
Hi

Testing the level converter on a pin-by-pin basis is best.  Apply 5V on B1 and get 3.3v on A1, then apply 0v on B1 and get 0v on A1. Then reverse the test for the A Side using 3.3v and 0v. 

Also check that there is 3.3v on OE. If not, the levels will not convert.  That said, I suspect your level converter is working.

I’m also presuming that you have tested the web interface – easiest if you set a static IP address (in config.h, lines 106-116) and browsing to that address on your WIFI system. If you can do that, the other parts of the system are working.

If that works, then the remaining issues are the circuit interface and the selection of the correct Pool Model in the Defs.  Unless all of that works, there is no point in connecting to the pool.

**Reply by visualapproach** [reply]
I'd go with the red LLC and only use filters to the display. And make the hardware test according to the build instructions.

**Reply by franpalacio** [reply]
I have already tried with the red LLC and the filters on the display side, I have taken the opportunity to try all the combinations that have occurred to me, with filters, without filters, with MIAMI2021, with PRE2021. And keep doing the same. The music plays and shows the IP and date on the display, which makes me think that the pins are correct and the level converter works, at least in part.
The hardware test result is correct:
CIO to DSP pin 3 success!CIO to DSP pin 4 success!CIO to DSP pin 5 success!CIO to DSP pin 3 success!CIO to DSP pin 4 success!CIO to DSP pin 5 success!
However, with the blue LLC, when trying to do the hardware test, the blue le remains fixed and does not start. But I have already tried 4 different LLC's and bought from 2 different sites and it does the same thing.
I am going to focus on the red LLC that seems to work at least the hardware test

**Comment by JPMNZ** [comment]
Hi

Yes, there are some other frustrations as well. The system seems to be unresponsive after a time – resetting the pump makes the Web Interface responsive again, but gradually, the responsiveness deteriorates. I think a watchdog timer of some sort might be needed. The Network “wake-up”  becomes unreliable on my system.

It may also be that there is a hardware difference between the S100100 and S100103 pumps – the latter works and the former is giving “franpalacio” no joy.

Regds

**Comment by visualapproach** [comment]
Just checking - did you define the correct PCB?
Otherwise - I don't know

**Comment by franpalacio** [comment]
Yes, I think so, I have defined PCB_V2 and the board is the one that appears in the photo.
In any case, I have tested everything, also changing the definition of the board and even crossing the cables as in #define PCB_V2B, in some of these tests I have broken the esp, and I have had to change it for another.
Could you do some low-level debugging to see what is being received by the pins in some way? more ideas? Have they been able to make any changes to the model of the pump?
Thank you

**Comment by franpalacio** [comment]
After doing many tests, I have already lost count, I don't know the atrocities I have done..., and in the absence of trying more things, I have managed to make it work as follows:
1 you connect the pump and after the music and welcome message, you connect the usb to the computer and you restart the serial output on platform.
It gives a few beeps and the screen turns on. It seems that it responds to everything both from the physical keys and from the web interface.
If you connect a usb charger, it doesn't work.
If you don't give the monitor task restart, it doesn't work either.
Any idea what's going on.
How can I solve it in a more elegant way?

by the way I am using the red LLC
Thanks

**Reply by visualapproach** [reply]
@franpalacio Sorry for not answering in a while, busy times. Watch out for the magic smoke when connecting more than one power source (USB + pump 5v) If they differ there can be current flowing between them uncontrolled. Depends on the ESP design, but be careful and check for hot parts on the ESP. That said, it sounds like some signal issue that happens to be on the right side of working when you connect the USB. The protocol (model/pcb) is clearly correctly set since it works after your funny procedure.
Have you tried without any LLC as suggested here: #312 ?

**Reply by franpalacio** [reply]
Don't worry, and thank you very much for your time.
Yes, I've tried bridging cables without the LLC and it doesn't work. The only way is the way I mentioned above.

**Comment by franpalacio** [comment]
I've been doing more crazy tests.
1 Turn on the pump, wait for the welcome message to appear, connect the usb cable from the esp8266 to the computer, press the serial monitor button in platformio (and if necessary restart the task if the pop-up window appears). At this point everything starts to work. Up to here is the proof of the previous message. Now I try to disconnect the usb cable, and everything still works correctly.
This has made me think that maybe it is some signal that the esp8266 receives from the computer when opening the serial monitor.
Then I have connected another esp8266 programming in the loop reads and writes in the serial port and I have connected it via usb to the esp8266 of the pump.
As a power source I have used the one that comes through the usb and also an external one to the vin and gnd.
In this second test it has not worked, it has not done anything.
I don't know if it could be related to the interruptions.
Previously, before arriving at this solution, by putting checkpoints in the code, the code inside the CIO interrupts would not execute.
Do the serial port pins have an interrupt associated with them that is activated when I connect the computer?
I don't know, if something occurs to you... I'll put the tests that occur to me here.
Regards

**Reply by visualapproach** [reply]
> This has made me think that maybe it is some signal that the esp8266 receives from the computer when opening the serial monitor.

The ESP (code) does not receive any messages from serial input. It only outputs messages (6-wire). The CIO CS and Data Pins are triggered by interrupts, but it has nothing to do with the USB serial being connected. I don't know how you tested it, but don't put Serial.print()'s within the interrupt routines. Change the state of a global volatile variable and print it in a regular function like .loop().
So, the USB trick is not doing anything to the program. I wonder if it just helps the ESP go through the WI-FI setup without browning out. You could try commenting out the wi-fi start function and see if it makes a difference.
What happens if you start up the pump with the USB serial monitor active already? What's the output?

**Reply by franpalacio** [reply]
Hello, I know that the CIO does not receive anything by Serial, I have not used the Serial.println() functions within the interruptions either, as checkpoints I was using the beep() function, uncommenting its content. That's why everything is so weird and I'm so baffled. I have tried what you have told me and I will post the results.
//startWiFi()0;
It does the same.

With the esp clicked on the pump:
1 I connect the usb of the computer
2 I hit the serial monitor button on the platform and I hope the task will be done. Meanwhile, the esp is already starting with the power of the PC and the music and welcome message come out
3 I connect the pump.
The output from the serial monitor is:

 *  Executing task in folder 6-wire-version develop: C:\Users\Fran\.platformio\penv\Scripts\platformio.exe device monitor --environment nodemcuv2 

--- Terminal on COM7 | 115200 8-N-1
--- Available filters and text transformations: colorize, debug, default, direct, esp8266_exception_decoder, hexlify, log2file, nocontrol, printable, send_on_enter, time
--- More details at https://bit.ly/pio-monitor-filters
--- Quit: Ctrl+C | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H
OTA > ready
HTTP > server started
WebSocket > server started
Failed to read mqtt.json. Using defaults.
MQTT > Buffer size successfully increased
Wrote cmdq.txt
{"LEN":0}
Wrote cmdq.txt
{"LEN":0}


and the pump in this way also works correctly.
I think it works as soon as the serial monitor task is executed in platformio
It's all very strange.
With another project that strange things happened to me at the end was that I had to change the number of a timer.
The same could be something similar, that there is a timer or something else that when receiving information through the serial from the PC changes something and it works. but i'm lost
I have also taken the opportunity to try another new ESP8266 that has come to me from another manufacturer and the same thing happens

**Reply by visualapproach** [reply]
Strange. Don't know what's happening

**Reply by franpalacio** [reply]
I have bought other LCC net on aliexpress. I'm waiting for them to arrive. I already told you.
greetings and thanks

**Comment by franpalacio** [comment]
hello again, the new red LCC has arrived, I have also taken the opportunity to solder a new board in case there were any failures and the behavior remains the same.
I'm thinking of adding checkpoints in the code to see how far it goes and where it gets stuck.
At the moment I have observed that the interruptions are called.
Any idea what order to follow to do these checks.
Also for future versions you could include a Debug mode
All the best

**Comment by visualapproach** [comment]
Connect according to pcb v2b for best chance of success.
About debugging, yes I've made a couple of debugging versions before but it should probably be done in a more intelligent way.

**Comment by franpalacio** [comment]
Hello, after trying all the possible combinations and in the absence of further testing, I just tried one more time and it works!!! Although I had already tried this configuration and it didn't work, now miraculously it has started to work. The only change that I have made, and that it occurs to me that it could be, has been to change the usb cable that I have used for programming. The cable that I used previously in theory worked correctly and I have programmed many other ESPs. So I don't understand it, but it's the only change I've made.
The configuration that I have used has been:
MIAMI2021
PCB_V2B

**Comment by KrustyMaz** [comment]
Hey
Thanks for this amazing idea, coding and introductions.
Even I have not managed to make it work yet. 

I have exactly this same kind of problem behavior with my pump that franpalacio.
Have tried to solve this with many tricks before I found this discussion.
Setup:
**Pump: S100101 date: 2022-01-14**
Pool: Hawaii without hydrojets
PCB: **BWC v2**
LLC: txs0108e blue

Configured/defined in firmware: MIAMI2021 and PCB_V2

Tested with connections/parts:
-	Wires to H1A + H2A
-	Wires to H1A + H2A + 10K resistor on 3V
-	Wires to H1A + H2B with 560ohm resistors DSP side only + 10K resistor on 3V
-	Wires to H1B with 680ohm resistors + H2B with 560ohm resistors DSP side + 10K resistor on 3V
(I have only few resistors to use and don’t have any capacitors to use/test)

Tested with webui hardwaretest and it seems ok.
Tested connections with multimeter.

Tested with another NodeMCU V3 chip and txs0108e blue LLC but those are exactly same manufacturers and versions etc

I get tune on when powering on, DSP shows IP/date, WebUI running ok, 
but temperature is not showing or buttons on webui or physical does nothing.

**So what else can I try to solve this problem?**
1.	PCB_V2B with crossed wires?
Like franpalacio mentioned that MIAMI2021+PCB_V2B configuration was successful.
_”On some 2021+ models you will need to use the red LLC and define the PCB_V2B in model.h”_
So because I don’t have this red LLC and only the blue one, I would need to order red one and wait for that to test. 
So blue LLC is not an option with this PCB_V2B connection type? 

2.	Test those blue LLCs are working ok with 5V, 3V power supply and multimeter?

3.	Try with new code version 4.10?
Although I think that this is connection old signal -problem
I used code version 3.5.1

4.	Method without LLC? like here: #312
They use Wemos D1 mini, but because I have theses PCB_V2 boards, would it be same if I just use my nodeMCU V3
Configure code same. define MIAMI2021 + PCB_V2
And only remove LLC and solder jumpwires to LLC pin holes like A1-B1, A2-B2 etc
And use H1A+H2A to CIO/DSP wires?

Regards
KrustyMaz

**Comment by JPMNZ** [comment]
Hi folks

Even though I authored this thread, I can report that some time ago I got the system working as I said in an earlier post, but not as reliably as I’d wish. It would sometimes fail to come up when browsed to on its IP address. I found this frustrating and was not prepared to trust it.

However, since constraining the MCU to a 2.4GHz WiFi net with no other traffic on it (separate SSID) and accessing it via a VPN, I have 99% reliability (just the odd failure of the HTML code to resolve the first time). I get the impression that the MCU is susceptible to too much WiFi traffic – including the hack attacks all routers can suffer from. The VPN and strong Router security appears to have fixed this.

I now control the Spa from anywhere and can trust it to do as it is instructed. So: for me, Case Closed.

Kind regards

John M

**Reply by KrustyMaz** [reply]
Because this conversation thread @JPMNZ started is in category Ideas and @franpalacio 's and my problem seems to be little offtopic, should I start new conversation thread to category Q&A with my explanations and tests so other users may easier find it?

**Reply by visualapproach** [reply]
Maybe do that if you succeed, so there is anything useful to find.

**Comment by visualapproach** [comment]
> Hey
> 
> Thanks for this amazing idea, coding and introductions.

You are welcome

> 
> Setup:
> 
> **Pump: S100101 date: 2022-01-14**
> Pool: Hawaii without hydrojets
> PCB: **BWC v2**
> LLC: txs0108e blue
> 
> Configured/defined in firmware: MIAMI2021 and PCB_V2
> **So what else can I try to solve this problem?**
> 
> 1.	PCB_V2B with crossed wires?

Yes

> Like franpalacio mentioned that MIAMI2021+PCB_V2B configuration was successful.
> 
> _”On some 2021+ models you will need to use the red LLC and define the PCB_V2B in model.h”_
> 
> So because I don’t have this red LLC and only the blue one, I would need to order red one and wait for that to test. 
> 
> So blue LLC is not an option with this PCB_V2B connection type? 

Yes it is but it depends on the pump. It seems more people have luck with the red one. 

> 
> 2.	Test those blue LLCs are working ok with 5V, 3V power supply and multimeter?

Doesn't hurt, but I'd try 4 first.

> 3.	Try with new code version 4.10?
> 
> Although I think that this is connection old signal -problem
> 
> I used code version 3.5.1

It shouldn't be the problem

> 4.	Method without LLC? like here: #312
> 
> They use Wemos D1 mini, but because I have theses PCB_V2 boards, would it be same if I just use my nodeMCU V3

Yes try this!


> Configure code same. define MIAMI2021 + PCB_V2
> 
> And only remove LLC and solder jumpwires to LLC pin holes like A1-B1, A2-B2 etc
> 
> And use H1A+H2A to CIO/DSP wires?

Yes. Or connect the wires directly to left side of where the LLC were. Do whatever is easiest for you.

> Regards
> 
> KrustyMaz


Also try to start the esp first with a usb cable and then start the pump. Some people has found it to work.

**Reply by KrustyMaz** [reply]
Problem seems to be solved

I tried first one and that solved the problem.
So I used PCB_V2**B with crossed wires** and **blue LLC**.
Only 10k resistor on 3V. No other parts.

At first try, pump CIO connection was ok and WEBUI worked ok, but DSP went crazy.
That problem solved with resoldering and shorter wires. 20cm wires to about 6cm wires.

So if someone has same kind of problems, you should try
PCB_V2B + LLC (blue txs0108e / YF08E was ok)
10K resistor on 3V
short wires

And when this was ok with 3.5.1 version, changed to 4.1.0 and everything OK with that too.
Thank you very much @visualapproach 
Coffee will be sent

**Reply by visualapproach** [reply]
Thanks for the write-up on your road to success! And for the coffee! Cheers

**Comment by janmittner** [comment]
I have 2021 Palm Springs with S100101 pump model. I also faced problems when following default tutorial. Problems were solved by using PCB V2B with crossed wires. Using red LLC, no resistors, long wires. So owners of S100101 should probably directly follow the V2B part with crossed wires.  

Thank you all for your work and information.

**Reply by Remigiusz2020** [reply]
hi. anyone want to sell v2b board?
